### PR TITLE
Added initial Intel SYCL support

### PR DESCRIPTION
**Pull request by ArindamK92**
Edited the below files to add new PPM for SYCL. Currently, it can detect Intel SYCL compiler
CMakeLists.txt
benchmarks/CMakeLists.txt
rosetta/src/CMakeLists.txt
rosetta/src/rosetta-common.cpp
rosetta/src/rosetta-stat.cpp
rosetta/src/rosetta/evaluator.py
rosetta/src/rosetta/scripts/gen-stage1.py

Added a dummy SYCL benchmark file:
benchmarks/idioms/assign/assign.sycl.cxx
benchmarks/idioms/pointwise/pointwise.sycl.cxx

Edited the below files to make them compatible with icpx compiler
rosetta/src/const.c
rosetta/src/gamma.c
rosetta/src/igam.c
rosetta/src/incbet.c
rosetta/src/incbi.c
rosetta/src/mtherr.c
rosetta/src/ndtr.c
rosetta/src/ndtri.c
rosetta/src/polevl.c
rosetta/src/stdtr.c

### CODE REVIEW

**Comment by Meinersbur** [comment]
So far, `assign.sycl.cxx` does not use any SYCL. Consider adding a minimal SYCL kernel that can be executed in a GPU.

File: rosetta/src/rosetta-stat.h [hunk 1]
```diff
@@ -3,16 +3,32 @@
...
+// #include "igam.cxx"
+// #include "incbet.cxx"
+// #include "incbi.cxx"
+// #include "mtherr.cxx"
+// #include "ndtr.cxx"
+// #include "ndtri.cxx"
+// #include "polevl.cxx"
+// #include "stdtr.cxx"
+
+extern "C++" {
```
Comment (Meinersbur) [inline rosetta/src/rosetta-stat.h:17]: Please remove `extern "C++"`. There is no need in an C++-only source.

File: rosetta/src/CMakeLists.txt [hunk 2]
```diff
@@ -2,16 +2,16 @@ set(_common_sources
...
-    "ndtr.c"
-    "ndtri.c"
-    "incbet.c"
-    "incbi.c"
-    "mtherr.c"
-    "const.c"
-    "gamma.c"
-    "polevl.c"
-    "igam.c")
+    "stdtr.cxx"
```
Comment (Meinersbur) [inline rosetta/src/CMakeLists.txt:5]: Could you use the extension `.cpp` like the other files in the directory?

File: rosetta/src/rosetta-stat.h [hunk 3]
```diff
@@ -3,16 +3,32 @@
...
 // From cephes library
 double stdtr(int k, double t);
 double stdtri(int k, double p);
 double ndtri(double y0);
 double ndtr(double a);
 }
 
+// extern "C" {
```
Comment (Meinersbur) [inline rosetta/src/rosetta-stat.h:25]: Please avoid adding commented-out code where feasible

File: benchmarks/idioms/pairwise/pairwise.sycl.cxx [hunk 4]
```diff
@@ -0,0 +1,43 @@
+// BUILD: add_benchmark(sycl)
+#include "rosetta.h"
+#include <CL/sycl.hpp>
+
+
+
+
+static void kernel(int n, double *C, double *B, double *A) {
```
[the remark this answers is not in the document]
Comment (ArindamK92) [reply benchmarks/idioms/pairwise/pairwise.sycl.cxx]: I found the pairwise benchmarks are actually not running and not using state class. So, I thought it is not required in the benchmark and deleted the pairwise.sycl.cxx file. Do you want me to update all of them and add the sycl code?

File: benchmarks/idioms/assign/assign.sycl.cxx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#include <rosetta.h>
+#include <CL/sycl.hpp> 
+
+
+static void kernel_test(pbsize_t n, real data[]) {
+  cl::sycl::queue q;
+  cl::sycl::buffer<real, 1> buf(data, cl::sycl::range<1>(n));
+
+    q.submit([&](cl::sycl::handler& cgh) {
+        auto acc = buf.get_access<cl::sycl::access::mode::read_write>(cgh);
```
[the remark this answers is not in the document]
Comment (ArindamK92) [reply benchmarks/idioms/assign/assign.sycl.cxx]: Fixed: Changed the access mode to write.

File: rosetta/src/CMakeLists.txt (hunk 2 above)
Comment (ArindamK92) [reply rosetta/src/CMakeLists.txt]: Fixed

File: rosetta/src/rosetta-stat.h (hunk 3 above)
Comment (ArindamK92) [reply rosetta/src/rosetta-stat.h]: Done

**Pull request merged**